apps/shelter/src/pages/manage/attendance/index.tsx

```diff
@@ -10,16 +11,123 @@ import {
   Thead,
   Tr,
 } from '@chakra-ui/react';
+import {
+  queryOptions,
+  useMutation,
+  useQueryClient,
+  useSuspenseQuery,
+} from '@tanstack/react-query';
+import { ChangeEvent, Suspense, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import {
+  getShelterApprovedRecruitmentApplicants,
+  updateAttendanceAPI,
+} from '@/apis/recruitment';
+import { AttendanceStatus } from '@/types/apis/recruitment';
+
+const attendanceQueryOptions = (recruitmentId: number) =>
+  queryOptions({
+    queryKey: ['attendance', recruitmentId],
+    queryFn: () => getShelterApprovedRecruitmentApplicants(recruitmentId),
+    select: ({ data }) => data,
+    refetchOnReconnect: false,
+    refetchOnWindowFocus: false,
+    refetchInterval: false,
+  });
+
+type Gender = 'MALE' | 'FEMALE';
+
+type Applicant = {
+  volunteerId: number;
+  applicantId: number;
+  volunteerName: string;
+  volunteerBirthDate: string;
+  volunteerGender: Gender;
+  volunteerPhoneNumber: string;
+  volunteerAttendance: boolean;
+};
+
+function AttendanceForm() {
+  const { id } = useParams() as { id: string };
```

Comment:
```suggestion
  const { id } = useParams<{ id: string }>();
```
generics를 활용하는게 더 좋을 것 같습니다!!